scripts/zones/Halvung/npcs/Operating_Lever_A.lua

```diff
@@ -0,0 +1,28 @@
+-----------------------------------
+-- Area: Halvung
+--  NPC: Operating Lever A
+-- TODO: more than 5/6 people still need verification as no sites show this requirment?
+-----------------------------------
+local ID = require("scripts/zones/Halvung/IDs")
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+end
+
+entity.onTrigger = function(player, npc)
+    if player:hasKeyItem(xi.ki.BRACELET_OF_VERVE) then
+        GetNPCByID(npc:getID() - 2):openDoor(30)
```

Comment:
This seems potentially brittle.  I think I'd feel better defining the ID in IDs.lua under the npc section, that way if anything is inserted between in the future we're safe.

Reply:
Would you kindly elaborate a little bit on brittle? 

MINING = 
{
}

KEYITEM = 
{
BRACELET_OF_VERVE = 2579
}

assuming that is for if ID's shift at some point?

Reply:
@claywar So thinking on this a little bit, this specific scenario has 10 levers, which when triggered open up the door (which is its own npcid). Would it be more efficient, long term, to name the doors by ID?
Example: 1st set of doors
Operating_Lever_A = 17031670 
Operating_Lever_B = 17031671
when triggered both those doors will open 17031668 (_1qf) 

Second set of doors
Operating_Lever_A = 17031674 
Operating_Lever_B = 1703167
when triggered both those doors will open 17031672 (_1qf) 

etc..

With that said, I could rename all the operating levers to Operating_Lever_A-J in npc_list.sql, define each actual door in IDs.lua (DOOR_A_B),  = 17031668, DOOR_C_D = 17031672 etc., and add lua for each of the Operating_Lever_A-J, to open the corresponding door?

`INSERT INTO `npc_list` VALUES (17031668,'_1qf','_1qf',0,-120.000,17.500,-180.000,1,40,40,9,0,0,0,6147,0x0200000000000000000000000000000000000000,0,'TOAU',0);
INSERT INTO `npc_list` VALUES (17031669,'_1qg','     ',0,-120.000,18.400,-183.292,1,40,40,9,0,0,0,6147,0x0200000000000000000000000000000000000000,0,'TOAU',0);
INSERT INTO `npc_list` VALUES (17031670,'Operating_Lever_A','Operating Lever',0,-117.819,18.349,-183.096,32769,40,40,0,0,0,0,3,0x0000340000000000000000000000000000000000,0,'TOAU',1);
INSERT INTO `npc_list` VALUES (17031671,'Operating_Lever_B','Operating Lever',0,-122.163,18.349,-183.100,32769,40,40,0,0,0,0,3,0x0000340000000000000000000000000000000000,0,'TOAU',1);
INSERT INTO `npc_list` VALUES (17031672,'_1qh','_1qh',0,-240.000,17.500,-140.000,1,40,40,9,0,0,0,6147,0x0200000000000000000000000000000000000000,0,'TOAU',0);
INSERT INTO `npc_list` VALUES (17031673,'_1qi','     ',0,-240.000,18.400,-143.292,1,40,40,9,0,0,0,6147,0x0200000000000000000000000000000000000000,0,'TOAU',0);
INSERT INTO `npc_list` VALUES (17031674,'Operating_Lever_A','Operating Lever',0,-237.813,18.347,-143.092,32769,40,40,0,0,0,0,3,0x0000340000000000000000000000000000000000,0,'TOAU',1);
INSERT INTO `npc_list` VALUES (17031675,'Operating_Lever_B','Operating Lever',0,-242.183,18.347,-143.092,32769,40,40,0,0,0,0,3,0x0000340000000000000000000000000000000000,0,'TOAU',1);
INSERT INTO `npc_list` VALUES (17031676,'_1qj','_1qj',0,-240.000,17.500,-20.000,1,40,40,9,0,0,0,6147,0x0200000000000000000000000000000000000000,0,'TOAU',0);
INSERT INTO `npc_list` VALUES (17031677,'_1qk','     ',0,-240.000,18.400,-16.707,1,40,40,9,0,0,0,6147,0x0200000000000000000000000000000000000000,0,'TOAU',0);
INSERT INTO `npc_list` VALUES (17031678,'Operating_Lever_A','Operating Lever',0,-237.804,18.339,-16.900,32769,40,40,0,0,0,0,3,0x0000340000000000000000000000000000000000,0,'TOAU',1);
INSERT INTO `npc_list` VALUES (17031679,'Operating_Lever_B','Operating Lever',0,-242.166,18.339,-16.900,32769,40,40,0,0,0,0,3,0x0000340000000000000000000000000000000000,0,'TOAU',1);
INSERT INTO `npc_list` VALUES (17031680,'_1ql','_1ql',0,-120.000,-12.500,220.000,1,40,40,9,0,0,0,6147,0x0200000000000000000000000000000000000000,0,'TOAU',0);
INSERT INTO `npc_list` VALUES (17031681,'_1qm','     ',0,-120.000,-11.600,223.292,1,40,40,9,0,0,0,6147,0x0200000000000000000000000000000000000000,0,'TOAU',0);
INSERT INTO `npc_list` VALUES (17031682,'Operating_Lever_A','Operating Lever',0,-117.805,-11.658,223.100,32769,40,40,0,0,0,0,3,0x0000340000000000000000000000000000000000,0,'TOAU',1);
INSERT INTO `npc_list` VALUES (17031683,'Operating_Lever_B','Operating Lever',0,-122.175,-11.658,223.100,32769,40,40,0,0,0,0,3,0x0000340000000000000000000000000000000000,0,'TOAU',1);
INSERT INTO `npc_list` VALUES (17031684,'_1qn','_1qn',0,-380.000,-2.499,240.000,1,40,40,9,0,0,0,6147,0x0200000000000000000000000000000000000000,0,'TOAU',0);
INSERT INTO `npc_list` VALUES (17031685,'_1qo','     ',0,-376.707,-1.599,240.000,1,40,40,9,0,0,0,6147,0x0200000000000000000000000000000000000000,0,'TOAU',0);
INSERT INTO `npc_list` VALUES (17031686,'Operating_Lever_A','Operating Lever',0,-376.898,-1.655,242.185,32769,40,40,0,0,0,0,3,0x0000340000000000000000000000000000000000,0,'TOAU',1);
INSERT INTO `npc_list` VALUES (17031687,'Operating_Lever_B','Operating Lever',0,-376.898,-1.655,237.835,32769,40,40,0,0,0,0,3,0x0000340000000000000000000000000000000000,0,'TOAU',1);`

Reply:
> This seems potentially brittle.  I think I'd feel better defining the ID in IDs.lua under the npc section, that way if anything is inserted between in the future we're safe.

If you mean by a gap (inserted) in between the ids, there isn't one as there is another npcid that is actually in the door frame but does nothing. ?? Thoughts

Reply:
npcs get shifted every month lately. It isn't a matter of all shifting +-X. Npcs get thrown in the middle for no reason.

With that in mind, I agree the NPCs should be defined in zones/Halvung/IDs file and just call them like you call text IDs